packages/editor-ui/src/stores/logStreaming.store.ts

```diff
@@ -32,221 +33,238 @@ export interface DestinationSettingsStore {
-			this.eventNames.clear();
-		},
-		addSelectedEvent(id: string, name: string) {
-			this.items[id]?.selectedEvents?.add(name);
-			this.setSelectedInGroup(id, name, true);
-		},
-		removeSelectedEvent(id: string, name: string) {
-			this.items[id]?.selectedEvents?.delete(name);
-			this.setSelectedInGroup(id, name, false);
-		},
-		getSelectedEvents(destinationId: string): string[] {
-			const selectedEvents: string[] = [];
-			if (this.items[destinationId]) {
-				for (const group of this.items[destinationId].eventGroups) {
-					if (group.selected) {
-						selectedEvents.push(group.name);
-					}
-					for (const event of group.children) {
-						if (event.selected) {
-							selectedEvents.push(event.name);
-						}
-					}
+			items[destination.id]?.selectedEvents?.clear();
+			if (destination.subscribedEvents) {
+				for (const eventName of destination.subscribedEvents) {
+					items[destination.id]?.selectedEvents?.add(eventName);
 				}
 			}
-			return selectedEvents;
-		},
-		setSelectedInGroup(destinationId: string, name: string, isSelected: boolean) {
-			if (this.items[destinationId]) {
-				const groupName = eventGroupFromEventName(name);
-				const groupIndex = this.items[destinationId].eventGroups.findIndex(
-					(e) => e.name === groupName,
-				);
-				if (groupIndex > -1) {
-					if (groupName === name) {
-						this.$patch((state) => {
-							state.items[destinationId].eventGroups[groupIndex].selected = isSelected;
-						});
-					} else {
-						const eventIndex = this.items[destinationId].eventGroups[groupIndex].children.findIndex(
-							(e) => e.name === name,
-						);
-						if (eventIndex > -1) {
-							this.$patch((state) => {
-								state.items[destinationId].eventGroups[groupIndex].children[eventIndex].selected =
-									isSelected;
-								if (isSelected) {
-									state.items[destinationId].eventGroups[groupIndex].indeterminate = isSelected;
-								} else {
-									let anySelected = false;
-									for (
-										let i = 0;
-										i < state.items[destinationId].eventGroups[groupIndex].children.length;
-										i++
-									) {
-										anySelected =
-											anySelected ||
-											state.items[destinationId].eventGroups[groupIndex].children[i].selected;
-									}
-									state.items[destinationId].eventGroups[groupIndex].indeterminate = anySelected;
-								}
-							});
+			items[destination.id].eventGroups = eventGroupsFromStringList(
+				eventNames.value,
+				items[destination.id]?.selectedEvents,
+			);
+		}
+	};
+
+	const getDestination = (destinationId: string) => {
+		if (items[destinationId]) {
+			return items[destinationId].destination;
+		} else {
+			return;
+		}
+	};
+
+	const getAllDestinations = () => {
+		const destinations: MessageEventBusDestinationOptions[] = [];
+		for (const key of Object.keys(items)) {
+			destinations.push(items[key].destination);
+		}
+		return destinations;
+	};
+
+	const clearDestinations = () => {
+		Object.assign(items, {});
```

Comment:
This doesn't actually clear anything. It assigns an empty object into items, which results into items staying the same.

```js
> var items = { a: 1, b: 2 }
undefined
> Object.assign(items, {})
{ a: 1, b: 2 }
> items
{ a: 1, b: 2 }
```

Reply:
Good catch! 

https://github.com/vuejs/core/issues/1081

Reply:
🤯 that is so confusing syntax and looks like it's wrong. Glad you changed it 👍